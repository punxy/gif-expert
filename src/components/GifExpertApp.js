import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)
   
    // const handleAdd = () => {
    //     //setCategories([ ...categories, 'HunterXHunter' ]);
    //     setCategories( cats =>  [...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />

            <AddCategory setCategories={setCategories} />

            <ol>
                { categories.map( category => 
                        <GifGrid key={category} category={category} />
                )}
            </ol>
        </>
    )
}
