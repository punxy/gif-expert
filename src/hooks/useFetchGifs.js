import { useEffect, useState } from "react"
import { getFetchGifs } from "../helpers/getFetchGifs"

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getFetchGifs(category)
            .then( images => {
                setState({
                    data: images,
                    loading: false
                })
            })
    }, [category]) 

    return state;
}