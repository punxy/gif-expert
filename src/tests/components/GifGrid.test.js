import React from 'react';
import { shallow } from "enzyme";
import '@testing-library/jest-dom';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Test in GifGrid component', () => {

    const category = 'Hunter x';
    const initState = {
        data: [],
        loading: true
    };

    test('should be render correctly', () => {
        useFetchGifs.mockReturnValue(initState);
        const wrapper = shallow( <GifGrid category={category} /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a paragraph with the title', () => {
        useFetchGifs.mockReturnValue(initState);
        const wrapper = shallow( <GifGrid category={category} /> );
        const h3 = wrapper.find('h3');
        expect(h3.text().trim()).toBe(category);
    });

    test('should display items when ages are loaded with useFetchGifs', () => { 

        const imgs = [
            {
                id: 'ABC',
                url: 'https://localhost/img.jpg',
                title: 'title example'
            },
            {
                id: '123',
                url: 'https://localhost/img.jpg',
                title: 'title example'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={category} /> );
        expect(wrapper.find('p').exists() ).toBeFalsy();
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    });
});