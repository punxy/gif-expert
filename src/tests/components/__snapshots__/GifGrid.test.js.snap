// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGrid component should be render correctly 1`] = `
<Fragment>
  <h3>
    Hunter x
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
